src/main/java/com/datatorrent/netlet/DefaultEventLoop.java

```diff
@@ -323,7 +323,12 @@ public void run()
             }
           }
           else {
-            register(channel, SelectionKey.OP_CONNECT, l);
+            /*
```

Comment:
I am also looking at this:
http://stackoverflow.com/questions/6540346/java-solaris-nio-op-connect-problem



On Tue, Jul 7, 2015 at 10:24 AM, Vlad Rozov <notifications@github.com>
wrote:

> In src/main/java/com/datatorrent/netlet/DefaultEventLoop.java
> <https://github.com/DataTorrent/Netlet/pull/3#discussion_r34065982>:
>
> > @@ -323,7 +323,12 @@ public void run()
> >              }
> >            }
> >            else {
> > -            register(channel, SelectionKey.OP_CONNECT, l);
> > +            /*
>
> I guess it fires OP_CONNECT or OP_CONNECT and OP_READ as adding OP_READ
> fixes the problem and OP_CONNECT is removed from interestOps.
>
> I ran few dozens tests where ClientListener.read() checks that
> ClientListener.connected() was called and that was always the case. It is
> hard to say whether during those tests the original issue was triggered or
> not. Usually it is triggered once in 10-20 runs, but adding more logging
> changes how often it reproduces.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/DataTorrent/Netlet/pull/3/files#r34065982>.
>